### PR TITLE
Migrate to Spring Cloud Build's checkstyle mechanism

### PR DESCRIPTION
**Pull request by marcingrzejszczak**
This is a PR with the setup for checkstyle and code formatting of all the other Spring Cloud projects. If you're interested in following our rules, don't hestitate to merge it (once you fix the checkstyle violations - if there are any).

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -49,6 +49,12 @@
 		<zipkin-gcp.version>0.9.0</zipkin-gcp.version>
 		<app-engine-maven-plugin.version>1.3.2</app-engine-maven-plugin.version>
 		<kotlin.version>1.3.10</kotlin.version>
+
+		<maven-checkstyle-plugin.failsOnError>true</maven-checkstyle-plugin.failsOnError>
+		<maven-checkstyle-plugin.failsOnViolation>true
+		</maven-checkstyle-plugin.failsOnViolation>
+		<maven-checkstyle-plugin.includeTestSourceDirectory>true
+		</maven-checkstyle-plugin.includeTestSourceDirectory>
```
Comment (artembilan) [inline pom.xml]: Some strange code formatting. Are we pursuing 90 length line now?

File: pom.xml [hunk 2]
```diff
@@ -147,28 +153,6 @@
...
-				<executions>
-					<execution>
-						<id>checkstyle-validation</id>
-						<phase>validate</phase>
-						<goals>
-							<goal>check</goal>
-						</goals>
-						<configuration>
-							<!-- Checkstyle rules inherited from spring-cloud-build -->
-							<suppressionsLocation>src/checkstyle/checkstyle-suppressions.xml</suppressionsLocation>
```
Comment (artembilan) [inline pom.xml]: I'm pretty sure we are still in the situation when we would like to suppress some rules.
So, what is new way to do that?

File: pom.xml [hunk 3]
```diff
@@ -204,6 +196,10 @@
 					<header>${project.name}</header>
 				</configuration>
 			</plugin>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-checkstyle-plugin</artifactId>
```
Comment (artembilan) [inline pom.xml]: Why do we need to duplicate this plugin?

File: pom.xml (hunk 2 above)
Comment (marcingrzejszczak) [reply pom.xml]: we automatically pick `root/src/checkstyle/checkstyle-suppressions.xml` file, so you don't need to explicitly provide it. All is described here - https://github.com/spring-cloud/spring-cloud-build#checkstyle

File: pom.xml (hunk 3 above)
Comment (marcingrzejszczak) [reply pom.xml]: Where do we duplicate it?

File: pom.xml (hunk 1 above)
Comment (marcingrzejszczak) [reply pom.xml]: It should be 120. Maybe I copied it improperly.

File: pom.xml [hunk 4]
```diff
@@ -185,6 +169,14 @@
 				<artifactId>appengine-maven-plugin</artifactId>
 				<version>${app-engine-maven-plugin.version}</version>
 			</plugin>
+			<plugin>
+				<groupId>io.spring.javaformat</groupId>
+				<artifactId>spring-javaformat-maven-plugin</artifactId>
+			</plugin>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-checkstyle-plugin</artifactId>
```
Comment (artembilan) [inline pom.xml]: > Where do we duplicate it?

Here

**Comment by artembilan** [comment]
BTW, we can't merge this because built fails with your changes. Therefore master is not going to be deployed because of your change.
If you introduce them, please, fix all the failures during build.

File: pom.xml (hunk 3 above)
Comment (marcingrzejszczak) [reply pom.xml]: One is in the reporting section and the other is in the main one

**Comment by marcingrzejszczak** [comment]
>  If you introduce them, please, fix all the failures during build.

No way :) I'm just suggesting what you can do to follow our approach. If you don't want to do that, that's not a problem but I', not going to fix your checkstyle issues.

**Comment by marcingrzejszczak** [comment]
> BTW, we can't merge this because built fails with your changes. Therefore master is not going to be deployed because of your change.

You can take over this branch, fix the issues and then push it back to the branch and then merge it. For sure I will not fix your checkstyle issues. If you don't like this approach I guess you can close this PR and continue working the way you do until now.

**Comment by artembilan** [comment]
```
[ERROR] Failed to execute goal io.spring.javaformat:spring-javaformat-maven-plugin:0.0.6:validate (default) on project spring-cloud-gcp-autoconfigure: Formatting violations found in the following files:
[ERROR]  * /home/travis/build/spring-cloud/spring-cloud-gcp/spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/sql/GcpCloudSqlAutoConfiguration.java
[ERROR]  * /home/travis/build/spring-cloud/spring-cloud-gcp/spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/storage/GcpStorageAutoConfiguration.java
```

You know, this is not a Checkstyle issue: the formatter one.
Since this is fully new stuff for us, would you mind to fix violation introduced by your change and guide for the proper way to format code in the future. And I guess it is going to be applied immediately to all upcoming changes just after merging your PR.

Thanks for understanding!

File: pom.xml (hunk 3 above)
Comment (artembilan) [reply pom.xml]: Right. I understand the layout. Only my concern that I don't see answer to the question "Why?".

> 5. Add checkstyle plugin to your build and reporting phases

Is not a reason to do this.
Why that reporting can't get appropriate plugins from the main section?

However what I see, according the plugin official docs, we just don't need it in the main section: https://maven.apache.org/plugins/maven-checkstyle-plugin/usage.html

That's how I am 🤷‍♂️ 

Thanks for understanding.

Comment (marcingrzejszczak) [reply pom.xml]: Maybe I misconfigured it then. If you think that it's wrong then let's change it.

**Comment by marcingrzejszczak** [comment]
Fixed the formatting violations.

**Comment by artembilan** [comment]
> Files changed 542 

Wat? How is that possible? What have you changed? 
I may guess that the story is about the mentioned *formatter*.
OK. let it be reformatted. We can live with that and really follow its requirements.

However we have a new problem:
```
[ERROR] Failed to execute goal org.jetbrains.kotlin:kotlin-maven-plugin:1.3.10:compile (compile) on project spring-cloud-gcp-kotlin-app-sample: Compilation failure: Compilation failure: 
[ERROR] /home/travis/build/spring-cloud/spring-cloud-gcp/spring-cloud-gcp-kotlin-samples/spring-cloud-gcp-kotlin-app-sample/src/main/kotlin/com/example/app/Controller.kt:[36,38] Unresolved reference: PubSubTemplate
[ERROR] /home/travis/build/spring-cloud/spring-cloud-gcp/spring-cloud-gcp-kotlin-samples/spring-cloud-gcp-kotlin-app-sample/src/main/kotlin/com/example/app/PubSubConfiguration.kt:[60,68] Unresolved reference: JacksonPubSubMessageConverter
[ERROR] /home/travis/build/spring-cloud/spring-cloud-gcp/spring-cloud-gcp-kotlin-samples/spring-cloud-gcp-kotlin-app-sample/src/main/kotlin/com/example/app/PubSubConfiguration.kt:[61,16] Unresolved reference: JacksonPubSubMessageConverter
[ERROR] /home/travis/build/spring-cloud/spring-cloud-gcp/spring-cloud-gcp-kotlin-samples/spring-cloud-gcp-kotlin-app-sample/src/main/kotlin/com/example/app/PubSubConfiguration.kt:[67,29] Unresolved reference: PubSubTemplate
[ERROR] /home/travis/build/spring-cloud/spring-cloud-gcp/spring-cloud-gcp-kotlin-samples/spring-cloud-gcp-kotlin-app-sample/src/main/kotlin/com/example/app/PubSubConfiguration.kt:[67,46] Unresolved reference: PubSubInboundChannelAdapter
[ERROR] /home/travis/build/spring-cloud/spring-cloud-gcp/spring-cloud-gcp-kotlin-samples/spring-cloud-gcp-kotlin-app-sample/src/main/kotlin/com/example/app/PubSubConfiguration.kt:[69,23] Unresolved reference: PubSubInboundChannelAdapter
[ERROR] /home/travis/build/spring-cloud/spring-cloud-gcp/spring-cloud-gcp-kotlin-samples/spring-cloud-gcp-kotlin-app-sample/src/main/kotlin/com/example/app/PubSubConfiguration.kt:[71,27] Unresolved reference: AckMode
[ERROR] /home/travis/build/spring-cloud/spring-cloud-gcp/spring-cloud-gcp-kotlin-samples/spring-cloud-gcp-kotlin-app-sample/src/main/kotlin/com/example/app/PubSubConfiguration.kt:[79,21] Unresolved reference: GcpPubSubHeaders
[ERROR] /home/travis/build/spring-cloud/spring-cloud-gcp/spring-cloud-gcp-kotlin-samples/spring-cloud-gcp-kotlin-app-sample/src/main/kotlin/com/example/app/PubSubConfiguration.kt:[79,21] An annotation argument must be a compile-time constant
[ERROR] /home/travis/build/spring-cloud/spring-cloud-gcp/spring-cloud-gcp-kotlin-samples/spring-cloud-gcp-kotlin-app-sample/src/main/kotlin/com/example/app/PubSubConfiguration.kt:[79,65] Unresolved reference: BasicAcknowledgeablePubsubMessage
```

And i still can guess that this result comes from the `formatter` as well. And looks like it doesn't know how to reformat Kotlin code properly. 
Unfortunately I can't see your fixes here in the PR, but this again comes as a result of your changes.

All my these nit-picks not about to reject your PR or give your some headache: I just want to help to make process easier and much cleaner since it feels like tools are not stable enough.
I remember your sample about Groovy script reformatting 😉

**Comment by meltsufin** [comment]
The auto-fixes seem reasonable. It looks like we just need to figure out what to do with those Kotlin errors.

**Comment by marcingrzejszczak** [comment]
Yeah maybe i did sth to the Kotlin files that i don't know about. I don't really know kotlin so i don't know what i did wrong. Feel free to check it out and update the intellij formatter in spring cloud build tools.

**Comment by marcingrzejszczak** [comment]
I can imagine that the same thing happened with Kotlin as it did with Groovy. i could have accidentally remove imports (cause intellij assumed they are unused). Also i saw that you're using static blocks that initialize the static fields. The formatter changes the order if fields and static blocks. I fixed it in one place by disabling the formatter whatsoever. Maybe i missed another spot.

**Comment by dzou** [comment]
I think it will take us some time to migrate to spring-cloud-build style correctly.

While the autoformatter resolves our formatting violations, they made many files less readable - like the newlines/breaks for many of the files do not seem quite correct. Our team will probably discuss the best way to include these fixes in and whether we ought to move in this direction (since it looks like it is going to be an involved task).

**Comment by meltsufin** [comment]
@marcingrzejszczak @artembilan I've updated this PR and everything seems to compile now. However, the main problem I see is that there is enforcement of a very short line length, definitely not 120 characters. Is this a new rule for Spring style?

File: spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/core/cloudfoundry/GcpCloudFoundryEnvironmentPostProcessor.java [hunk 5]
```diff
@@ -78,27 +78,39 @@
...
-				.put("ProjectId", "project-id")
-				.put("PrivateKeyData", "credentials.encoded-key")
-				.build()),
-		TRACE("google-stackdriver-trace", "trace", new MapBuilder<String, String>()
-				.put("ProjectId", "project-id")
-				.put("PrivateKeyData", "credentials.encoded-key")
-				.build()),
-		MYSQL("google-cloudsql-mysql", "sql", sqlPropertyMap),
-		POSTGRES("google-cloudsql-postgres", "sql", sqlPropertyMap);
+				.put("PrivateKeyData", "credentials.encoded-key").build()), STORAGE(
```
Comment (meltsufin) [inline spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/core/cloudfoundry/GcpCloudFoundryEnvironmentPostProcessor.java]: This is a funny one.

**Comment by meltsufin** [comment]
Closing since this way out of date now.

**Pull request closed without merging**